### PR TITLE
Updated Mod To 1.20.1

### PR DESCRIPTION
**Pull request by DinnerBeef**
- Moved version numbers to gradle.properties
- Changed DiscriminatedName to Name as discord no longer has numbers
- Fixed AdvancementEvent to AdvancementEarnEvent
- Fixed the Name for AdvancementEarnEvent
- Added message to config to make sure 'Message Content Intent' is enabled for the bot

PS: This is the only image this could be tested with 
image

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -118,7 +118,11 @@ configurations {
 }
 minecraft.runs.all {
     lazyToken('minecraft_classpath') {
-        configurations.library.copyRecursive().resolve().collect { it.absolutePath }.join(File.pathSeparator)
+        configurations.library.copyRecursive().resolve()
```
Comment (Gegy) [inline build.gradle:121]: Hmm, I think that `minecraftLibrary` is the intended way to do this with newer ForgeGradle versions (in which case, these will need to be excluded in the dependency declaration) - can we move this there already, even if not upgrading FG yet?

File: src/main/java/org/jedi/DiscordMessageFormatter.java [hunk 2]
```diff
@@ -49,12 +49,13 @@ public Component format(Message message) {
 
     private Component formatAuthorName(Message message) {
         final TextColor color = this.getAuthorColor(message);
-        final MutableComponent discriminatedName = Component.literal(message.getAuthor().getDiscriminatedName());
+        System.out.println(message.getAuthor().getDisplayName());
```
Comment (Gegy) [inline src/main/java/org/jedi/DiscordMessageFormatter.java]: Debug?

File: build.gradle [hunk 3]
```diff
@@ -36,7 +36,7 @@ minecraft {
     //
     // Use non-default mappings at your own risk. They may not always work.
     // Simply re-run your setup task after changing the mappings to update your workspace.
-    mappings channel: 'official', version: '1.19'
+    mappings channel: 'official', version: "${minecraft_version}"
```
Comment (Gegy) [inline build.gradle]: Nit:
```suggestion
    mappings channel: 'official', version: minecraft_version
```

**Review by Gegy — COMMENTED** [review]
Looks great, mostly just small Gradle things 😄

File: src/main/java/org/jedi/DiscordMessageFormatter.java (hunk 2 above)
Comment (DinnerBeef) [reply src/main/java/org/jedi/DiscordMessageFormatter.java]: Lol must of missed one

**Comment by DinnerBeef** [comment]
I fixed what I could. I don't know too much about FrorgeGradle, so I have idea how you would go about fixing that one part.

**Review by Gegy — APPROVED** [review]

**Pull request merged**